src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalyzerExtensions.cs

```diff
@@ -13,10 +14,11 @@ public static class DiagnosticAnalyzerExtensions
         /// <param name="compilation">Compilation to which analyzers are to be added.</param>
         /// <param name="analyzers">The set of analyzers to include in future analyses.</param>
         /// <param name="options">Options that are passed to analyzers.</param>
+        /// <param name="onAnalyzerException">Action to invoke if an analyzer throws an exception.</param>
         /// <param name="cancellationToken">A cancellation token that can be used to abort analysis.</param>
-        public static CompilationWithAnalyzers WithAnalyzers(this Compilation compilation, ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerOptions options = null, CancellationToken cancellationToken = default(CancellationToken))
+        public static CompilationWithAnalyzers WithAnalyzers(this Compilation compilation, ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerOptions options = null, Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException = null, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
This also requires a new overload rather than adding a parameter--changing the existing API is not allowed.